text/0000-Project-Goals-2025h2.md

```diff
@@ -0,0 +1,754 @@
+* The work on [evolving trait hierarchies](https://rust-lang.github.io/rust-project-goals/2025h2/./evolving-traits.html) will make it possible to refactor some parts of an existing trait out into a new supertrait so they can be used on their own. This unblocks a number of features where the existing trait is insufficiently general, in particular stabilizing support for custom receiver types, a prior project goal that wound up blocking on this refactoring.
+* The work to [expand Rust's `Sized` hierarchy](https://rust-lang.github.io/rust-project-goals/2025h2/./scalable-vectors.html) will permit us to express types that are neither `Sized` nor `?Sized`, such as extern types (which have no size) or ARM's Scalable Vector Extensions (which have a size that is known at runtime, but not compilation time). This goal builds on [RFC #3729](https://github.com/rust-lang/rfcs/pull/3729)[] and [RFC #3838](https://github.com/rust-lang/rfcs/pull/3838)[], authored in previous project goal periods.
+* [In-place initialization](https://rust-lang.github.io/rust-project-goals/2025h2/./in-place-initialization.html) allows creating structs and values that are tied to a particular place in memory. While useful directly for projects doing advanced C interop, it also unblocks expanding `dyn Trait` to support for `async fn` and `-> impl Trait` methods, as compiling such methods requires the ability for the callee to return a future whose size is not known to the caller.
+
+### "Flexible, fast(er) Rust builds"
+
+| Goal                                                                | Point of contact | Team(s) and Champion(s)                                      |
+| :--                                                                 | :--         | :--                                                          |
+| [build-std](https://rust-lang.github.io/rust-project-goals/2025h2/build-std.html)                                           | [David Wood][]  | [cargo] ([Eric Huss][]), [compiler] ([David Wood][]), [libs] ([Amanieu d'Antras][]) |
+| [Promoting Parallel Front End](https://rust-lang.github.io/rust-project-goals/2025h2/parallel-front-end.html)               | [Sparrow Li][] | [compiler]                                                   |
+| [Production-ready cranelift backend](https://rust-lang.github.io/rust-project-goals/2025h2/production-ready-cranelift.html) | [Folkert de Vries][] | [compiler], [wg-compiler-performance]                        |
+
+
+The "Flexible, fast(er) Rust builds" initiative focuses on improving Rust's build system to better serve both specialized use cases and everyday development workflows:
```

Comment:
What does "Rust's build system" mean? It sounds like the goal is to improve bootstrap :sweat_smile: Maybe something like "focuses on improving compilation of Rust programs ..."?

Reply:
Cargo is rust's canonical build system. Absolutely no doubt in my mind it means "improving cargo"

Reply:
Let me rephrase for greater clarity!